### PR TITLE
Add optional "same branch only" input

### PR DESCRIPTION
**Pull request by gavinsharp**
This adds a `same-branch-only` input which can be set to `false` if you want to wait on pending runs regardless of which branch they're from.

I was trying to use this action to prevent simultaneous deploys, but realized it would only work to prevent deploys from the same branch. In our setup all non-main branches deploy to the same environment, so I need to wait on any pending run, even if it is from a different branch.

(Even more ideal would be a way to say "wait for all main or non-main branches, depending on whether this is a run from main or non-main", but that seemed too complex and also maybe too unique to our particular deploy setup to be worth supporting.)

This parameter defaults to true to preserve existing behaviour.

### CODE REVIEW

**Comment by gavinsharp** [comment]
It looks like passing `undefined` as the branch param to `OctokitGitHub.runs` is not behaving the way I would expect (finds no runs rather than finds matching runs). Let me find a way to fix that.

**Comment by gavinsharp** [comment]
> It looks like passing `undefined` as the branch param to `OctokitGitHub.runs` is not behaving the way I would expect (finds no runs rather than finds all runs). Let me find a way to fix that.

Fixed in 53c876b2606a1fb26a3811f8dc045e729d865c58.

File: src/input.ts [hunk 1]
```diff
@@ -21,6 +22,9 @@ export const parseInput = (env: Record<string, string | undefined>): Input => {
   const continueAfterSeconds = env["INPUT_CONTINUE-AFTER-SECONDS"]
     ? parseInt(env["INPUT_CONTINUE-AFTER-SECONDS"], 10)
     : undefined;
+  const sameBranchOnly =
+    env["INPUT_SAME-BRANCH-ONLY"] === "true" ||
+    env["INPUT_SAME-BRANCH-ONLY"] === undefined; // true if not specified
```
Comment (softprops) [inline src/input.ts:27]: Could we keep the default the same and make this an opt in? That would make this a non breaking change for people that depend on the existing behavior.

Comment (softprops) [reply src/input.ts:27]: Oh nevermind that _is_ the current behavior

**Comment by gavinsharp** [comment]
I was trying to figure out a way to write a test to capture the actual functionality better, but since the underlying behaviour that matters (filtering by branch) lives in Octokit, testing a mocked Octokit didn't seem that useful.

**Comment by softprops** [comment]
I'm sorry. I've let this one go. If you'd still like this functionality can you rebase with latest version of the code? I can commit to taking a look this weekend.

**Comment by gavinsharp** [comment]
> I'm sorry. I've let this one go. If you'd still like this functionality can you rebase with latest version of the code? I can commit to taking a look this weekend.

No worries. I've merged in master (just required a rebuild for `dist/index.js`, merged cleanly otherwise)

**Comment by gavinsharp** [comment]
I lied: trivial fix also needed for the new test (935b7e2d1d229431633b8b156a80dd660ebc012c)

**Comment by douglascamata** [comment]
I'm waiting for this, thanks! <3

**Comment by chenrui333** [comment]
This would be a great addon for terraform operations (which relies on shared resources)

**Comment by softprops** [comment]
I'll update the tag for this today

**Comment by softprops** [comment]
updated the latest tag with this change

**Pull request merged**